### PR TITLE
Make entity ids lowercase

### PR DESCRIPTION
**Pull request by grkvlt**
Needed to allow use of entity and application ids as part of DNS names, which
are case-insensitive. Extension to 10 characters gives equivalent amount of
randomness, so collision probability is unchanged.

### CODE REVIEW

**Comment by grkvlt** [comment]
Here are the relevant combinations for before:
```
62^8  =   218 340 105 584 895
```
And possibilities considered for after:
```
36^8  =     2 821 109 907 456
36^9  =   101 559 956 668 416
36^10 = 3 656 158 440 062 976
```

**Comment by neykov** [comment]
LGTM, but note that `makeRandomPassword` has a "special" behaviour in that it will pick initial characters from all pools and only after that merge them and select randomly regardless of the pool. Might be better to extend `makeRandomId` to accept character pools, otherwise all IDs will always follow the format <letter><number><letter or number>{8}

**Comment by grkvlt** [comment]
@neykov I like your suggestion of overloading `makeRandomId()` so I will implement that instead

**Comment by grkvlt** [comment]
@neykov I have updated the `Identifiers#makeRandomId(int)` to have the overload you suggested

File: utils/common/src/main/java/org/apache/brooklyn/util/text/Identifiers.java [hunk 1]
```diff
@@ -87,29 +87,34 @@
...
+     * which computed a random every 3 turns -- takes about 600 ns to do id
+     * of len 10, compared to 10000 ns for old version [on 1.6ghz machine]
+     */
+    public static String makeRandomId(int l, String validStartChars, String validNonStartChars) {
+        if (l <= 0) return "";
         char[] id = new char[l];
-        int d = random.nextInt( (26+26) * (26+26+10) * (26+26+10) * (26+26+10) * (26+26+10));
+        int s = validStartChars.length();
+        int n = validNonStartChars.length();
+        int d = random.nextInt(s * n * n * n * n);
```
Comment (neykov) [inline utils/common/src/main/java/org/apache/brooklyn/util/text/Identifiers.java]: This will overflow for very low `n` (~73).

Comment (grkvlt) [inline utils/common/src/main/java/org/apache/brooklyn/util/text/Identifiers.java]: @neykov Good catch, suppose we could change to a long, or reduce to re-randomising after less characters...

Comment (grkvlt) [inline utils/common/src/main/java/org/apache/brooklyn/util/text/Identifiers.java]: I'm going to use groups of 4 characters, and will limit the character set strings to a length of 128 maximum. Do you think it should error on too many characters, or silently truncate to the first 128?

Comment (neykov) [inline utils/common/src/main/java/org/apache/brooklyn/util/text/Identifiers.java]: Failing on too many characters is better I think
One more option - if input is too long switch to a slower implementation, but one which can handle the input correctly.
For this case though suggest making the parametrized `makeRandomId` private, adding the proper javadoc warnings and adding another method for lower case IDs. It requires the least effort and is good enough for now.

**Comment by grkvlt** [comment]
@neykov I added a note on overflow and made the method private

**Comment by neykov** [comment]
Thanks @grkvlt, merging.

**Pull request merged**